src/common/session/session-entity.js

```diff
@@ -0,0 +1,219 @@
+import { generateRandomHexString } from '../ids/unique-id'
+import { warn } from '../util/console'
+import { stringify } from '../util/stringify'
+import { ee } from '../event-emitter/contextual-ee'
+import { Timer } from '../timer/timer'
+import { isBrowserScope } from '../util/global-scope'
+import { DEFAULT_EXPIRES_MS, DEFAULT_INACTIVE_MS, PREFIX } from './constants'
+import { LocalMemory } from '../storage/local-memory'
+import { InteractionTimer } from '../timer/interaction-timer'
+import { wrapEvents } from '../wrap'
+
+export class SessionEntity {
+  /**
+   * Create a self-managing Session Entity. This entity is scoped to the agent identifier which triggered it, allowing for mutliple simultaneous session objects to exist.
+   * The key is inteded to be unique, if not, existing entities in the same agent ID will overlap and overwrite.
```

Comment:
This is a remnant comment that has no meaning now.  Basically we need to be aware that -- as of right now -- there is one "namespace" an agent can store data in LS.  If there are two agents on one page, they could overwrite each other since they would both use the same namespace in LS by default.

Reply:
example --
agent 1 --> NRBA_SESSION = abcd1234
agent 2 --> NRBA_SESSION = zyxw9876

One is gonna overwrite the other in that scenario, we just need to be aware of that